docs/why.qmd

```diff
@@ -2,24 +2,313 @@
+there have been projects attempting to scale the pandas API, they always result
+in a dubious support matrix. You can see that with
+[Modin](https://modin.readthedocs.io/en/stable/supported_apis/dataframe_supported.html)
+or [pandas on Spark (formerly known as
+Koalas)](https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/supported_pandas_api.html).
+
+[Google BigQuery
+DataFrames](https://github.com/googleapis/python-bigquery-dataframes) are a more
+modern attempt to scale the pandas API, built on top of Ibis. If you are going
+to build a pandas API, we recommend you take a look at this project.
+
+PySpark is a great API for Spark, but not very Pythonic and tightly coupled to
+the Spark execution engine.
+
+Ibis takes inspiration from pandas and PySpark -- and R and SQL -- but is designed to be
+scalable from the start. If offers a neutral,
+[self-governed](https://github.com/ibis-project/governance) open source option
+for your data platform.
+:::
+
+### Ibis for data organizations
+
+Data organizations can use Ibis to standardize the interface for SQL and Python
+data practitioners. It also allows them to:
+
+- transfer data between systems easily
+- transform, analyze, and prepare data where it lives
+- mix SQL and Python code seamlessly, with all the benefits of a general-purpose
+ programming language
+
+### Ibis for data library developers
+
+Python developers creating data libraries can use Ibis to:
+
+- instantly support 20+ data backends
+- instantly support pandas, PyArrow, and Polars objects
+- read and write from all common file formats (depending on the backend)
+- trace column-level lineage through Ibis expressions
+- compile Ibis expressions to SQL or Substrait
+- perform cross-dialect SQL transpilation (powered by
+ [SQLGlot](https://github.com/tobymao/sqlglot))
+
+## How does Ibis work?
 
 Most Python dataframes are tightly coupled to their execution engine. And many
 databases only support SQL, with no Python API. Ibis solves this problem by
 providing a common API for data manipulation in Python, and compiling that API
 into the backend's native language. This means you can learn a single API and
 use it across any supported backend (execution engine).
 
-![](./portability.png)
+Ibis supports three types of backend:
+
+1. SQL-generating backends
+2. Expression-generating backends
+3. Naïve execution backends
+
+```{python}
+# | code-fold: true
+import plotly.graph_objects as go
+
+backend_categories = {
+    "SQL-generating": [
+        "BigQuery",
+        "ClickHouse",
+        "DataFusion",
+        "Druid",
+        "DuckDB",
+        "Exasol",
+        "Flink",
+        "Impala",
+        "MSSQL",
+        "MySQL",
+        "Oracle",
+        "PostgreSQL",
+        "PySpark",
+        "RisingWave",
+        "Snowflake",
+        "SQLite",
+        "Trino",
+    ],
+    "Expression-generating": ["Dask", "Polars"],
+    "Naïve execution": ["pandas"],
+}
```

Comment:
For a script this long, maybe better served by pulling it out into a separate `.py` file and then importing and executing it in the qmd file?

As an aside, I wonder if we should add a category attribute (or something) to the backends that's one of "sql", "expression" or "naive".  Might be overkill, but we'll get tired of maintaining this list at some point.  (That can be for a followup, but I doubt this is the only place we'll find ourselves wanting to categorize the backends in)

Reply:
how do I execute it after importing? just something like:

````
```{python}
from whatever_script_thing import whatever

whatever()
```
````

I guess that would do it

Reply:
yes -- you might have to wrap up the `fig`-producing code in a function and have it return `fig`, then you can do
```
from whatever import whatever

fig = whatever()
fig
```

Reply:
I just did `from backend_sankey import fig` and `fig.show()`